src/include/sof/common.h

```diff
@@ -28,6 +28,8 @@
 	({const typeof(((type *)0)->member)*__memberptr = (ptr); \
 	(type *)((char *)__memberptr - offsetof(type, member)); })
```

Comment:
Please explain why is this the preferred way. Also, is there an instance of this in the code? I think it would be better to see at least few instances where we need to use the new macro.

Reply:
I'd love to see it used in trace macros and in the definition of SHARED_DATA, that's two places where you put stuff in different sections from the default one.

Reply:
Added

Reply:
Btw, GCC already does shorten some of the builtins, I'm assuming these are for XCC ?

Reply:
@dbaluta PR comment says Checkpatch wants the shortened version, and now there are examples of uses (hopefully all), is it okay now?

Reply:
@ktrzcinx fine for me now. I'm concerned about @lgirdwood comment. We should really check which shorten versions rare already supported by GCC and define them only for XCC.

Reply:
@lgirdwood @dbaluta I'm not aware of __attribute__((section)) being one of the builtins that is already shortened. From a quick test, gcc does not have this shortening by default.

Reply:
@lgirdwood I haven't seen this shorten names for GCC, but interesting. I get define name from checkpatch.pl:5922